### DISCUSSION TITLE
Description for url parameter

### DISCUSSION THREAD

**Discussion by jrkoiter** [apiflask/apiflask#361] [Q&A] [answered]
In the [basic example](https://github.com/apiflask/apiflask/blob/dd9c2bf51a8b2251c0970d68f727f5f5431d4ec9/examples/basic/app.py#L30), how would one add a description to the generated docs for the `pet_id` url parameter?

**Comment by jrkoiter** [comment]
Looks related to this issue: https://github.com/apiflask/apiflask/issues/350

**Reply by greyli** [reply]
Yes, I will merge #353 soon. After that, the schema (with field descriptions) for path location will overwrite the path variables defined in the URL rule.

**Comment by greyli** [comment, marked as answer]
Create a schema for all the path/URL parameters, then set the `input(location)` to `path`:

```py
class EnumPathParameter(Schema):
    image_type = String(validate=OneOf(['jpg', 'png', 'tiff', 'webp']))

@app.get('images/<image_type>')
    @app.input(EnumPathParameter, location='path')
    def index(image_type, path_data):
        pass
```
